src/modules/nodejs.rs

```diff
@@ -20,15 +19,19 @@ pub fn segment(context: &Context) -> Option<Segment> {
     match get_node_version() {
         Some(node_version) => {
             const NODE_CHAR: &str = "⬢";
-            const SEGMENT_COLOR: Color = Color::Green;
+            const MODULE_COLOR: Color = Color::Green;
 
-            let mut segment = Segment::new("node");
-            segment.set_style(SEGMENT_COLOR.bold());
+            let mut module = Module::new("node");
+            module.set_style(MODULE_COLOR.bold());
 
+            let symbol = module.new_segment("symbol");
+            symbol.set_value(NODE_CHAR);
```

Comment:
Since we are returning `self` with these methods we can use chaining to get rid of the intermediate vars. I think they distract a away from the builder pattern. we are really just mutating the module instance 👍 

```rust
module.new_segment("symbol").set_value(NODE_CHAR);
```

Reply:
I was thinking the same thing. That would mean that `new_segment()` would have to return a `&mut Segment` instead of a `Segment`, but we eventually need to return an owned `Segment`.

In the end it would look like this:
```rust
let module = module.new_segment("symbol").set_value(NODE_CHAR).build();

Some(module)
```

`.build()` would be returning the instance of `Segment`.

Reply:
On second thought, I could have the methods all pass around `mut Segment` instead of `&mut Segment`. 🤔
I wonder what the performance costs are.

Reply:
...But then we can't work with segments that were already added to `Module`s 
since we're no longer operating with references. 😖

Reply:
No... Those remain references so it's fine.

...

I'll do a little thinking without typing to GitHub

Reply:
hmmm yeah i misread i understand now. I was thinking we were passing the module reference around instead of pointers to its segments. 

I guess another approach is you could have add a second argument to new_segment method or have the setters on module rather than the module affix. Not sure how much value there is in doing this tho? 

``` rust
new_segment(name, value)
update_segment(name, value, style)

// which could eventually lead to something like:
let module = Module::new("rust")
                .set_style(MODULE_COLOR.bold())
                .new_segment("symbol", RUST_CHAR)
                .new_segment("version", format_rustc_version(rust_version)));
```
or we could return a pointer to the segment with `new_segment` and return a pointer to the module with the `set_value` but im not sure thats the best approach.

Reply:
Oh! You were talking specifically about the `new_segment()` methods. Got it! 😂

Reply:
pointers will probably best fast than passing structs around the call stack. My guess is that the differences are probably so minimal we wouldn't notice since they are not that big
.

Reply:
honestly i am fine with the current approach, just thought it could an interesting avenue to explore :

Reply:
There will be cases where segments will themselves have prefixes and suffixes, which is why we're setting the value separately.

Do you see this as an improvement?
```rust
const NODE_CHAR: &str = "⬢";
let module_color = Color::Green.bold();

let mut module = Module::new("node");
module.set_style(module_color);

let formatted_version = node_version.trim();
module.new_segment("symbol").set_value(NODE_CHAR);
module.new_segment("version").set_value(formatted_version);

Some(module)
```

Reply:
> There will be cases where segments will themselves have prefixes and suffixes, which is why we're setting the value separately.

I don't think i have a very clear picture of all the use cases but personally i feel its cleaner.

Reply:
We could certainly make `new_segment_with_value()` or something along those lines.

Reply:
> I don't think i have a very clear picture of all the use cases but personally i feel its cleaner.

To give context, here's an example of a module that will have a segment with a prefix and suffix:

```
[kubectl prefix] [kubectl symbol] [kubectl context] 
       at               ☸️                aks1

[kubectl namespace prefix] [kubectl namespace] [kubectl namespace suffix]
             (                 kube-system                  )
```
![image](https://user-images.githubusercontent.com/4658208/56613497-ba0eed00-65e4-11e9-9e20-937b4811a7ec.png)

To be honest, I may be creating this API a little too defensively. 😅

Reply:
I guess there won't ever be a situation where a segment _won't_ have a value.
What you're suggesting definitely seems like a better approach. 😄
I'll go ahead and add that.